src/components/Modals/Send/hooks/useFormSend/useFormSend.tsx

```diff
@@ -24,27 +26,50 @@ export const useFormSend = () => {
   const handleSend = async (data: SendInput) => {
     if (wallet) {
       try {
-        const adapter = chainAdapter.byChain(data.asset.chain)
+        const adapter = chainAdapterManager.byChain(data.asset.chain)
         const value = bnOrZero(data.crypto.amount)
           .times(bnOrZero(10).exponentiatedBy(data.asset.precision))
           .toFixed(0)
 
-        const { estimatedFees, feeType } = data
-        const fees = estimatedFees[feeType]
-        const fee = fees.feePerUnit
-        const gasLimit = fees.chainSpecific?.feeLimit
+        const adapterType = adapter.getType()
 
-        const accountType = allAccountTypes[getAccountTypeKey(data.asset.chain)]
-        const accountParams = accountType ? utxoAccountParams(data.asset, accountType, 0) : {}
-        const { txToSign } = await adapter.buildSendTransaction({
-          to: data.address,
-          value,
-          erc20ContractAddress: data.asset.tokenId,
-          wallet,
-          fee,
-          gasLimit,
-          ...accountParams
-        })
+        let result
+        if (adapterType === ChainTypes.Ethereum) {
+          const { estimatedFees, feeType } = data
+          const fees = estimatedFees[feeType] as FeeData<ChainTypes.Ethereum>
+          const fee = fees.feePerUnit
+          const gasLimit = fees.chainSpecific?.feeLimit
+          const accountType = allAccountTypes[getAccountTypeKey(data.asset.chain)]
+          const accountParams = accountType ? utxoAccountParams(data.asset, accountType, 0) : {}
+          result = await adapter.buildSendTransaction({
+            to: data.address,
+            value,
+            erc20ContractAddress: data.asset.tokenId,
+            wallet,
+            fee,
+            gasLimit,
+            ...accountParams
+          })
+        } else if (adapterType === ChainTypes.Bitcoin) {
+          const { estimatedFees, feeType } = data
+          const fees = estimatedFees[feeType] as FeeData<ChainTypes.Bitcoin>
+          const fee = fees.feePerUnit
+          const gasLimit = fees.chainSpecific?.byteCount
+          const accountType = allAccountTypes[getAccountTypeKey(data.asset.chain)]
+          const accountParams = accountType ? utxoAccountParams(data.asset, accountType, 0) : {}
+          result = await adapter.buildSendTransaction({
+            to: data.address,
+            value,
+            erc20ContractAddress: data.asset.tokenId,
+            wallet,
+            fee,
+            gasLimit,
+            ...accountParams
+          })
```

Comment:
Is there any common logic/data structures here that can be consolidated and minimize the `if (chain)` code.

Reply:
ya there is. good call

Reply:
done